src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -203,23 +206,38 @@ private static ServiceUnavailableRetryStrategy getServiceUnavailableRetryStrateg
       @Override
       public boolean retryRequest(
           final HttpResponse response, final int executionCount, final HttpContext context) {
-        int statusCode = response.getStatusLine().getStatusCode();
-        if (executionCount == MAX_RETRIES + 1) {
-          LOGGER.info("Reached the max retry time, not retrying anymore");
-          return false;
+        Object firstFault = context.getAttribute(FIRST_FAULT_TIMESTAMP);
+        long totalRetryDurationSoFarInSeconds = 0;
+        if (firstFault == null) {
+          context.setAttribute(FIRST_FAULT_TIMESTAMP, Instant.now());
+        } else {
+          Instant firstFaultInstant = (Instant) firstFault;
+          Instant now = Instant.now();
+          totalRetryDurationSoFarInSeconds = Duration.between(firstFaultInstant, now).getSeconds();
+
+          if (totalRetryDurationSoFarInSeconds > TOTAL_RETRY_DURATION.getSeconds()) {
+            LOGGER.info("Reached the max retry time, not retrying anymore");
```

Comment:
nit: could you add the max retry time to the message?

Reply:
Added